### DISCUSSION TITLE
Ruta REST de un formulario

### DISCUSSION THREAD

**Discussion by julchat** [dds-jv/foro#160] [Consultas de UI / REST] [answered]
Hola, estabamos practicando y nos surgió una duda, ¿Cuál seria la ruta correcta para un form que luego haga un POST?

Concretamente, en el parcial de QMP, pensamos para el formulario de creación de una prenda

GET /Usuario/:idUsuario/Prendas/new

es valido eso? o debería especificar mas el recurso que la operación que quiero hacer?
algo como
GET /Usuario/:idUsuario/Prendas/formulario

Gracias y saludos!

**Comment by atta25** [comment]
Buenas, me sumo con una consulta de Rutas del mismo parcial, adjunto link: [QueMePongo](https://docs.google.com/document/d/15ifQWQqfzbwwHIiiuoZiE4eyhd_DpmaQ5OGPvD9x2Wc/edit?usp=sharing)
Había pensado en esta solución:

> - Cargar una prenda:
Method: POST
/guardarropas/:id_guardarropa/prendas

Pero también había pensado esta otra opción:

> Method: POST
/prendas
Y el id del guardarropa lo mandaría por body junto con los datos de la prenda.

Cuál de las 2 sería más acertada?

Saludos.

**Comment by gastonprieto** [comment]
Hola @julchat!

Fijate que eso esta comentado en el documento que mando Franco al grupo hace poco y explica estos casos raros de cuando modelamos rutas HTTP

Hola @atta25!

Por como se viene dando el flujo de que primero se entra al guardarropas elegiria la opción 1), creo que es mas representativa del caso de uso, además esa tiene la ventaja de que el id del guardarropas si o si tiene que estar en la ruta.

La 2da es como mas abstracta y muestra una menor precedencia de objetos, aunque es valida también, pero creo que se adecua menos para este caso de uso, y deberías comentar todo lo que mandas en el body

**Comment by atta25** [comment]
Genial! Gracias por la respuesta.

**Comment by julchat** [comment, marked as answer]
Gracias! Se entendió todo!
